web/src/utils/UrlCheck.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ *
+ * @param link to check whether internal or external using regex
+ */
+export const isExternalUrl = (link: string): boolean => {
+  const domainRe = /^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/
+  return domainRe.test(link)
```

Comment:
What does this check for? That there is a host name? I don't really like this tbh. The behavior is very not clear and it does not work for internal links with a host name, e.g. https://integreat.app/augsburg. Instead I think something like the following makes sense:
```suggestion
const InternalLinkRegexp = new RegExp(buildConfig().internalLinksHijackPattern)

try {
  // Check whether link is a valid URL
  new URL(link)
  return !InternalLinkRegexp.test(link)
} catch (e) {
  // Link is not a valid URL and therefore just a pathname -> internal link
  return false
}
```

Reply:
But maybe I also don't get what that regexp is about :D

Reply:
your solution is fine :) its integrated